float32 percent
float32 remaining_distance
